### PR TITLE
specialHandling compiler "backdoor" works for models

### PR DESCRIPTION
**Pull request by perrydv**
This PR extends the “backdoor” or “specialHandling” system to models.  This system allows experimentation with rewriting C++ code and seeing if it will compile within the nimble workflow.  The compiled objects would be dangerous to use (could break if meaningful changes have been made), but it is useful to see if C++ compilation is successful in order to experiment with redesigning C++ model code.

[Documentation](https://github.com/nimble-dev/nimble/wiki/Developer-backdoor-to-manually-replace-generated-C#backdoor-for-models)

### CODE REVIEW

File: packages/nimble/R/nimbleProject.R [hunk 1]
```diff
@@ -508,8 +520,10 @@ nimbleProjectClass <- setRefClass('nimbleProjectClass',
                                      modelDef <- model$getModelDef()
                                      modelDefName <- modelDef$name
                                      Cname <- Rname2CppName(modelDefName)
-                                     if(is.null(filename)) filename <- paste0(projectName, '_', Rname2CppName(modelDefName)) 
-                                     
+                                     if(is.null(filename)) {
+                                         filename <- paste0(projectName, '_', Rname2CppName(modelDefName)) 
+                                         nfFileName <- paste0(projectName, '_', Rname2CppName(modelDefName),'_nfCode')
```
Comment (fritzo) [inline packages/nimble/R/nimbleProject.R]: Should `nfFileName` happen outside of this `if` statement? (I'm just confirming that you intended the behavior change.)

Comment (perrydv) [inline packages/nimble/R/nimbleProject.R]: Ah, you're right, it's an unintended change in behavior.

Comment (perrydv) [inline packages/nimble/R/nimbleProject.R]: Just fixed it.

**Comment by fritzo** [comment]
:+1:

**Comment by perrydv** [comment]
We are now timing out on Travis-CI due to tests running > 3 hours.  Let's discuss.

**Comment by fritzo** [comment]
I think it's safe to merge this if time-limit is the reason for build failure.

**Pull request merged**